hummingbot/client/performance.py

```diff
@@ -52,199 +54,234 @@ def __init__(self):
+        self.avg_s_price = self.divide(self.s_vol_quote, self.s_vol_base)
+        self.avg_tot_price = self.divide(abs(self.b_vol_quote) + abs(self.s_vol_quote),
+                                         abs(self.b_vol_base) + abs(self.s_vol_base))
+        self.avg_b_price = abs(self.avg_b_price)
+        self.avg_s_price = abs(self.avg_s_price)
+
+        return buys, sells
+
+    async def _calculate_fees(self, exchange: str, quote: str, trades: List[Any]):
+        for trade in trades:
+            if self._is_trade_fill(trade):
+                if trade.trade_fee.get("percent") is not None and trade.trade_fee["percent"] > 0:
+                    if quote not in self.fees:
+                        self.fees[quote] = s_decimal_0
+                    self.fees[quote] += Decimal(trade.price * trade.amount * trade.trade_fee["percent"])
+                for flat_fee in trade.trade_fee.get("flat_fees", []):
+                    if flat_fee["asset"] not in self.fees:
+                        self.fees[flat_fee["asset"]] = s_decimal_0
+                    self.fees[flat_fee["asset"]] += Decimal(flat_fee["amount"])
+            else:  # assume this is Trade object
+                if trade.trade_fee.percent > 0:
+                    if quote not in self.fees:
+                        self.fees[quote] = s_decimal_0
+                    self.fees[quote] += (trade.price * trade.order_amount) * trade.trade_fee.percent
+                for flat_fee in trade.trade_fee.flat_fees:
+                    if flat_fee[0] not in self.fees:
+                        self.fees[flat_fee[0]] = s_decimal_0
+                    self.fees[flat_fee[0]] += flat_fee[1]
+
+        for fee_token, fee_amount in self.fees.items():
+            if fee_token == quote:
+                self.fee_in_quote += fee_amount
+            else:
+                last_price = await get_last_price(exchange, f"{fee_token}-{quote}")
+                if last_price is not None:
+                    self.fee_in_quote += fee_amount * last_price
+
+    def _calculate_trade_pnl(self, buys: list, sells: list):
+        self.trade_pnl = self.cur_value - self.hold_value
+
+        # Handle trade_pnl differently for derivatives
+        if self._are_derivatives(buys) or self._are_derivatives(sells):
+            buys_copy, sells_copy = self.aggregate_position_order(buys.copy(), sells.copy())
+            long = []
+            short = []
+
+            while True:
+                lng = self.position_order(buys_copy, sells_copy)
+                if lng is not None:
+                    long.append(lng)
+
+                sht = self.position_order(sells_copy, buys_copy)
+                if sht is not None:
+                    short.append(sht)
+                if lng is None and sht is None:
+                    break
+
+            self.trade_pnl = Decimal(str(sum(self.derivative_pnl(long, short))))
+
+    async def _initialize_metrics(self,
```

Comment:
Missing type hint on returned value.

Reply:
This method initializes the internal status of the object, but it does not return anything.